### PR TITLE
Protocol changes for 1.21.40

### PR DESCRIPTION
**Pull request by dries-c**

### CODE REVIEW

**Review by nicholass003 — APPROVED** [review]

**Review by ShockedPlot7560 — APPROVED** [review]

File: src/ProtocolInfo.php [hunk 1]
```diff
@@ -32,11 +32,11 @@ private function __construct(){
 	 */
 
 	/** Actual Minecraft: PE protocol version */
-	public const CURRENT_PROTOCOL = 729;
+	public const CURRENT_PROTOCOL = 748;
 	/** Current Minecraft PE version reported by the server. This is usually the earliest currently supported version. */
-	public const MINECRAFT_VERSION = 'v1.21.30';
+	public const MINECRAFT_VERSION = 'v1.21.40';
```
Comment (AzaleeX) [inline src/ProtocolInfo.php:37]: I know it's not normally very useful but change to `1.21.41`

File: src/ProtocolInfo.php [hunk 2]
```diff
@@ -32,11 +32,11 @@ private function __construct(){
...
 	/** Version number sent to clients in ping responses. */
-	public const MINECRAFT_VERSION_NETWORK = '1.21.30';
+	public const MINECRAFT_VERSION_NETWORK = '1.21.40';
```
Comment (AzaleeX) [inline src/ProtocolInfo.php:39]: I know it's not normally very useful but change to 1.21.41

File: src/ProtocolInfo.php (hunk 1 above)
Comment (AzaleeX) [inline src/ProtocolInfo.php:37]: I know it's not normally very useful but change to 1.21.41

File: src/ProtocolInfo.php (hunk 2 above)
Comment (KnosTx) [reply src/ProtocolInfo.php:39]: That's not necessary because it's the same as 1.21.40 and 1.21.41

File: src/ProtocolInfo.php (hunk 1 above)
Comment (KnosTx) [reply src/ProtocolInfo.php:37]: That's not necessary because it's the same as 1.21.40 and 1.21.41

**Comment by Superomarking** [comment]
`PacketPool` changes are missing https://github.com/pmmp/BedrockProtocolDumper

**Review by TwistedAsylumMC — DISMISSED** [review]

**Review by dktapps — DISMISSED** [review]
Some autogen changes are missing

File: src/types/ServerAuthMovementMode.php [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+declare(strict_types=1);
+
+namespace pocketmine\network\mcpe\protocol\types;
+
+enum ServerAuthMovementMode : int{
+	use PacketIntEnumTrait;
+
+	case LEGACY_CLIENT_AUTHORITATIVE_V1 = 0;
+	case SERVER_AUTHORITATIVE_V2 = 1;
+	case SERVER_AUTHORITATIVE_V3 = 2;
```
[the remark this answers is not in the document]
Comment (dries-c) [reply src/types/ServerAuthMovementMode.php:22]: I'm aware, however this is now a Mojang defined enum, I'd like to kill PlayerMovementType in favour of this if everyone is fine with that.

[the remark this answers is not in the document]
Comment (dktapps) [reply src/types/ServerAuthMovementMode.php:22]: Well it was always Mojang defined. But regardless, do whatever you think is best, just as long as there aren't 2 of the same thing :P

**Comment by ClousCloud** [comment]
It's only been two days but it feels like two weeks

**Comment by KnosTx** [comment]
Why?

**Review by ClousCloud — APPROVED** [review]

**Review by dktapps — APPROVED** [review]

**Pull request merged**